gointelowl/gointelowl.go

```diff
@@ -0,0 +1,82 @@
+package gointelowl
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Struct for IntelOwl.
+// Initiate it using Token, Instance, Certificate
+type IntelOwlClient struct {
+	Token       string
+	URL         string
+	Certificate string
+}
+
+// Take url and IntelOwlClient and place request and return response
+func buildAndMakeGetRequest(url string, client IntelOwlClient) *http.Response {
+	request, err := http.NewRequest("GET", url, nil)
+	request.Header.Add("Authorization", "Token "+client.Token)
+	request.Header.Add("User-Agent", "IntelOwlClient/3.0.1")
+	if err != nil {
+		log.Fatalln(err)
+	}
+	httpClient := http.Client{}
+	response, err := httpClient.Do(request)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	log.Println(url, response.StatusCode)
+	return response
+}
+
+// Fetch list of all tags.
+//        Endpoint: ``/api/tags``
+// It Returns Slice of Tags
+func (client IntelOwlClient) GetAllTags() []map[string]string {
```

Comment:
we should pass the pointer to the client and not the instance so this should be `client *IntelOwlClient`.

Reply:
same change applies to other methods